drivers/modem/gsm_ppp.c

```diff
@@ -681,5 +694,32 @@ static int gsm_init(struct device *device)
 	return 0;
 }
 
+void restart_gsm_init(void)
+{
+	int r;
+
+	is_booting_phase = false;
+
+	/*
+	 * Switching to modem UART interface to enable the modem for CONNECT
+	 */
+
+	r = modem_iface_uart_init_dev(&gsm.context.iface,
+			CONFIG_MODEM_GSM_UART_NAME);
+
+
+	k_thread_create(&gsm_rx_thread, gsm_rx_stack,
```

Comment:
Could you create a separate function that would init the modem uart and start the thread. Then call that function here and from `gsm_init()` so we can share the code and save some flash.

Where / how the old thead is killed? We want to create the thread only once, what you are doing in application side when the restart needs to be done?

Reply:
if muxing is not enabled, the thread is aborted but now in new version I addressed with checks on MUX enable. rather gsm_configure is combined with thread rather than uart.